pkgs/servers/samba/4.x.nix

```
@@ -223,7 +223,7 @@ stdenv.mkDerivation rec {
   '';
 
   disallowedReferences =
-    lib.optionals (buildPackages.python3Packages.python != python)
+    lib.optionals (buildPackages.python3Packages.python != python3Packages.python)
       [ buildPackages.python3Packages.python ];
```

Rather than reverting this change, it might make sense to extend it to verify that we don't end up with two copies of Python (regular and libxcrypt-legacy) in the closure:
```suggestion
    lib.optional (buildPackages.python3Packages.python != python3Packages.python)
      buildPackages.python3Packages.python ++
    lib.optional (python3Packages.python != python) python3Packages.python;
```
```

---

Actually, if you try this you will discover that samba does have multiple Pythons in its closure. Fixing that might be out of scope for this PR though.